bottlerocket/samples/Makefile.toml

```diff
@@ -0,0 +1,242 @@
+CLUSTER_NAME = { script=["echo $(echo ${ARCHITECTURE} | tr '_' '-')-$(echo ${VARIANT} | tr -d '.')"], condition = { env_not_set = ["CLUSTER_NAME"] } }
+
+[tasks.ova-name.env]
+OVA_NAME = { script=["echo bottlerocket-${VARIANT}-x86_64-${STARTING_VERSION}.ova"], condition = { env_not_set = ["OVA_NAME"] } }
+
+[tasks.bottlerocket-ami-id.env]
+BOTTLEROCKET_AMI_ID = { script=['''
+if [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "migration" ]; then
+    AMI_VERSION=$(echo ${STARTING_VERSION} | tr -d "v")
+else
+    AMI_VERSION="latest"
+fi
+aws ssm get-parameter \
+    --region ${AWS_REGION} \
+    --name "/aws/service/bottlerocket/${VARIANT}/$(echo ${ARCHITECTURE} | sed -e 's/aarch64/arm64/g')/${AMI_VERSION}/image_id" \
+    --query Parameter.Value --output text 
+'''], condition = { env_not_set = ["BOTTLEROCKET_AMI_ID"], env_contains = { "VARIANT" = "aws" } } }
+
+[tasks.gpu.env]
+GPU = { script=['''
+if [ $(echo ${VARIANT} | tr "-" " " | awk '{print $NF}') = "nvidia" ]; then
+    echo true
+else
+    echo false
+fi
+'''], condition = { env_not_set = ["GPU"] } }
+
+[tasks.mgmt-cluster-kubeconfig-base64.env]
+MGMT_CLUSTER_KUBECONFIG_BASE64 = {script=['''
+if [ -n "${MGMT_CLUSTER_KUBECONFIG_PATH}" ]; then
+    cat ${MGMT_CLUSTER_KUBECONFIG_PATH} | base64
+fi
+''']}
+
+[tasks.instance-types]
+dependencies = ["gpu"]
+
+[tasks.instance-types.env]
+INSTANCE_TYPES = {script=['''
+if [ -z ${INSTANCE_TYPES} ]; then
+    if [ ${ARCHITECTURE} = "x86_64" ]; then
+        if [ ${GPU} = "true" ]; then
+            echo "[\"g4dn.xlarge\"]"
+        else
+            echo "[\"m5.large\"]"
+        fi
+    else
+        if [ ${GPU} = "true" ]; then
+            echo "[\"g5g.xlarge\"]"
+        else
+            echo "[\"m6g.large\"]"
+        fi
+    fi
+else
+    echo ${INSTANCE_TYPES}
+fi
+''']}
+
+[tasks.set-env]
+dependencies = ["set-agent-images", "cluster-name", "gpu", "metadata-url", "ova-name", "bottlerocket-ami-id", "mgmt-cluster-kubeconfig-base64", "instance-types"]
+
+[tasks.create-inner]
+dependencies = ["set-env"]
+script_runner = "bash"
+script = '''
+eval "cat > ${OUTPUT_FILE} << EOF
+$(< ${CLUSTER_TYPE}/${TEST_TYPE}.yaml)
+EOF
+" 2> /dev/null
+
+echo "${OUTPUT_FILE}"
+'''
+
+[tasks.create-inner.env]
+OUTPUT_FILE = {script=['''
+if [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "migration" ]; then
+    echo "output/${CLUSTER_NAME}-migration.yaml"
+elif [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "workload" ]; then
+    echo "output/${CLUSTER_NAME}-workload.yaml"
+else
+    echo "output/${CLUSTER_NAME}.yaml"
+fi
+''']}
+
+[tasks.create]
+script = '''
+#!@duckscript
+set_env TEST_TYPE ${1}
+cm_run_task ${1}
+cm_run_task create-inner
+'''
+
+[tasks.install-controller]
+dependencies = ["set-agent-images"]
+script = "cli install --controller-uri ${CONTROLLER_IMAGE_URI}"
+
+[tasks.add-aws-secret]
+dependencies = ["install-controller"]
+condition = { env_contains = { "CLUSTER_TYPE" = "kind", "VARIANT" = "aws" } }
```

Comment:
Will `"VARIANT" = "aws"` ever happen? We also need aws secrets for VMware migration testing. So I'd add the secret for all kind samples.

Reply:
NVM. I see that it is looking for a substring here.

Reply:
It's an `env_contains` target so this checks if VARIANT contains the string `aws` (the syntax is a bit confusing). I'm not sure where the aws secret for vmware migration testing would be used, since all migration testing samples are in the `eks` folder. Do you mean just in case someone adds a migration test file that would be running on a `kind` cluster?

Reply:
Just to be safe. Currently the migration agent requires aws secret.